### PR TITLE
Rework admin user and group setup using `api.admin` module

### PR DESCRIPTION
**Pull request by gctucker**
Rework the tool to bootstrap a new database by creating an admin user and group directly into any MongoDB instance while reusing the Authentication, Database and model classes from the API.  This doesn't require any service to be running apart from MongoDB.

### CODE REVIEW

File: api/auth.py [hunk 1]
```diff
@@ -39,8 +39,9 @@ class Authentication:
     should be a db.Database object.
     """
 
+    CRYPTCTX = CryptContext(schemes=["bcrypt"], deprecated="auto")
```
Comment (JenySadadia) [inline api/auth.py:42]: I'd suggest `CRYPT_CTX` for better readability.
```suggestion
    CRYPT_CTX = CryptContext(schemes=["bcrypt"], deprecated="auto")
```

File: api/admin.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        return None
+    password = getpass.getpass(f"Password for user '{args.username}': ")
+    hashed_password = Authentication.get_password_hash(password)
+    print(f"Creating {username} user...")
+    return await db.create(User(
+        username=username,
+        hashed_password=hashed_password,
+        is_admin=True,
+        groups=[admin_group]
+    ))
```
Comment (JenySadadia) [inline api/admin.py:45]: We have removed `is_admin` field.
```suggestion
    return await db.create(User(
        username=username,
        hashed_password=hashed_password,
        groups=[admin_group]
    ))
```

File: scripts/setup_admin_user [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+
```
Comment (JenySadadia) [inline scripts/setup_admin_user:4]: How about starting `db` service from the script itself?

File: api/admin.py (hunk 2 above)
Comment (gctucker) [reply api/admin.py:45]: Yes, that was before the changes were merged in the User schema.  I'll update it.

File: scripts/setup_admin_user (hunk 3 above)
Comment (gctucker) [reply scripts/setup_admin_user:4]: Because it's not necessarily for the `db` service, in fact I did this to set up a new database in Atlas.

Comment (JenySadadia) [reply scripts/setup_admin_user:4]: Oh, yes. Mongo URL can be anything and not just the local `db` service.

**Comment by gctucker** [comment]
I'll also make a separate PR to update the documentation about creating the initial admin user accordingly.

File: api/admin.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        username=username,
+        hashed_password=hashed_password,
+        groups=[admin_group]
+    ))
+
+
+async def main(args):
+    db = Database(args.mongo, args.database)
+    group = await setup_admin_group(db)
+    user = await setup_admin_user(db, args.username, group)
```
Comment (JenySadadia) [inline api/admin.py:50]: `pylint` is complaining about the unused `user` variable.

**Comment by JenySadadia** [comment]
Tested OK on staging.

**Comment by JenySadadia** [comment]
Also, need to drop https://github.com/kernelci/kernelci-api/blob/main/init_user_groups script.

**Comment by gctucker** [comment]
Right, I think that can be done as a follow-up.  The documentation also needs to be updated when removing the previous script.

File: api/admin.py (hunk 4 above)
Comment (JenySadadia) [reply api/admin.py:50]: This can be fixed in a follow-up PR as well. Merging it for now.

**Comment by JenySadadia** [comment]
> Right, I think that can be done as a follow-up. The documentation also needs to be updated when removing the previous script.

Sounds good.

**Pull request merged**